pkg/controller/logstorage/cleanup/cleanup_controller.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cleanup
```

Comment:
I deliberately didn't keep this as a strictly "users cleanup" controller because I figured there might be other things later we need to clean up and this way we can simply extend this controller.

Reply:
I think this is ever-so-slightly at odds with our original assertion that this controller is only needed for multi-tenant clusters, since "other cleanup" isn't necessarily limited to multi-tenant. I think it would be more clear if we made this either:

- A multi-tenant cleanup controller that only cleans up multi-tenant things
- A user cleanup controller, that only cleans up ES users.

My fairly strong preference is for the latter - In general I am in favor of small, purpose built controllers rather than controllers that try to do a lot of disjoint things. 

Similarly, the operator already performs lots of cleanup, so a generic cleanup controller is a bit at odds with the fact that it won't ever be the only place that cleanup of resources is performed.

Reply:
Renamed and moved this to an `esuserscleanup` controller.